src/azure-cli/azure/cli/command_modules/iot/custom.py

```diff
@@ -407,6 +428,9 @@ def iot_hub_create(cmd, client, hub_name, resource_group_name, location=None,
         raise CLIError('Key-based authentication requires a connection string.')
     if identity_based_file_upload and not fileupload_storage_container_uri:
         raise CLIError('Identity-based authentication requires a storage container uri (--fileupload-storage-container-uri, --fcu).')
+    if not identity_based_file_upload and fileupload_storage_identity:
+        raise CLIError('In order to set a fileupload storage identity, please set file upload storage authentication (--fsa) to IdentityBased')
```

Comment:
Please use the specific error type `RequiredArgumentMissingError` instead of CLIError.

Reply:
@zhoxing-ms Just for this new check? Or for all relevant CLIErrors in this block?